geocodingsearchv7/batchgeocoder_test.go

```diff
@@ -0,0 +1,108 @@
+package geocodingsearchv7_test
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"io"
+	"net/http"
+	"testing"
+
+	"go.einride.tech/here/geocodingsearchv7"
+	"gotest.tools/v3/assert"
+)
+
+type BatchGeocodingMock struct {
+	responseStatus int
+	responseBody   geocodingsearchv7.BatchGeocoderResponse
+}
+
+func (c *BatchGeocodingMock) Do(req *http.Request) (*http.Response, error) {
+	headers := http.Header{}
+	headers.Add("Content-Type", "application/json")
+	b, err := json.Marshal(c.responseBody)
+	if err != nil {
+		return nil, err
+	}
+	r := bytes.NewReader(b)
+	return &http.Response{
+		StatusCode:    c.responseStatus,
+		Header:        headers,
+		Body:          io.NopCloser(r),
+		ContentLength: int64(len(b)),
+	}, nil
+}
+
+func TestBatchGeocodingService_BatchGeocode(t *testing.T) {
+	t.Parallel()
+	ctx := context.Background()
+
+	t.Run(
+		"when both addresses and queries is nil, then return InvalidArgument ",
+		func(t *testing.T) {
+			t.Parallel()
+			httpClient := BatchGeocodingMock{responseBody: geocodingsearchv7.BatchGeocoderResponse{}, responseStatus: 200}
+			routingClient := geocodingsearchv7.NewClient(&httpClient)
+
+			_, err := routingClient.BatchGeocoder.BatchGeocoderUpload(ctx, &geocodingsearchv7.BatchGeocoderUploadRequest{
+				Addresses: nil, // Addresses is nil
+				Queries:   nil, // Queries is nil
```

Comment:
```suggestion
				Addresses: nil,
				Queries:   nil,
```
Do you think these comments might be superfluous if the test's name anyways indicates that the intention is for these fields to be nil?